nautilus_core/persistence/src/db/database.rs

```diff
@@ -106,30 +108,52 @@ impl Database {
     }
 }
 
-#[warn(dead_code)]
-#[cfg(test)]
-mod tests {
-    use std::path::Path;
-
-    use sqlx::{sqlite::SqliteConnectOptions, FromRow, Row, SqlitePool};
-
-    use crate::db::database::{Database, DatabaseEngine};
-    async fn setup_test_database() -> Database {
-        // check if test_db.sqlite exists,if not, create it
-        let db_path = std::env::var("TEST_DB_PATH").unwrap_or("test_db.sqlite".to_string());
-        let db_file_path = Path::new(db_path.as_str());
-        let exists = db_file_path.exists();
-        if !exists {
-            SqlitePool::connect_with(
-                SqliteConnectOptions::new()
-                    .filename(db_file_path)
-                    .create_if_missing(true),
-            )
-            .await
-            .expect("Failed to create test_db.sqlite");
+pub async fn init_db_schema(db: &Database, schema_dir: &str) -> Result<()> {
+    // scan all the files in the current directory
+    let mut sql_files =
+        std::fs::read_dir(schema_dir)?.collect::<Result<Vec<_>, std::io::Error>>()?;
+
+    for file in sql_files.iter_mut() {
+        let file_name = file.file_name();
+        println!("Executing SQL file: {:?}", file_name);
+        let file_path = file.path();
+        let sql_content = std::fs::read_to_string(file_path.clone())?;
+        for sql_statement in sql_content.split(';').filter(|s| !s.trim().is_empty()) {
+            db.execute(sql_statement).await.unwrap_or_else(|e| {
+                panic!(
+                    "Failed to execute SQL statement: {} with reason {}",
+                    file_path.display(),
+                    e
+                )
+            });
         }
-        Database::new(Some(DatabaseEngine::SQLITE), Some("sqlite:test_db.sqlite")).await
     }
+    Ok(())
+}
+
+pub async fn setup_test_database() -> Database {
```

Comment:
Consider using sqlx test attribute to setup test db for tests. https://docs.rs/sqlx/latest/sqlx/attr.test.html

Reply:
Yeah, but best to have these manual function where you have more control what are you doing. Also this function could will be exported with pyo3 so that python code can also use it; specially useful in pyo3 python tests